p2p/peer.py

```diff
@@ -121,7 +121,9 @@
          writer
          ) = await auth.handshake(remote, privkey, token)
     except (ConnectionRefusedError, OSError) as e:
-        raise UnreachablePeer(e)
+        raise UnreachablePeer() from e
+    except MalformedMessage as e:
```

Comment:
It turns out this is the wrong place to catch this exception; it is raised by `Peer.do_p2p_handshake()` and not `auth.handshake()`